modules/machine-health-checks-about.adoc

```diff
@@ -0,0 +1,29 @@
+// Module included in the following assemblies:
+//
+// * master/deploying-machine-health-checks.adoc
+
+[id='machine-health-checks-about-{context}']
+= About `MachineHealthChecks`
+
+`MachineHealthChecks` are a specialized implementation of the `MachineAutoscaler`
```

Comment:
`MachineHealthChecks` is unrelated to `MachineAutoscaler`. It's a feature built on top of the machine API

Reply:
As @enxebre says, `MachineHealthChecks` is consumed by machine health-check controller, `MachineAutoscaler` by cluster autoscaler operator. Completely independent of each other. `MachineAutoscaler` "just" annotates already existing `MachineSet` object so it can be recognized by the cluster autoscaler. In general, cluster autoscaler removes nodes that are underutilized. Compared to machine health-checker that replaces nodes that are (in a way) broken. It's not guarantied both autoscaler and health checker will not interfere with each other though. Most likely, when a node becomes NotReady and pods running on the node are killed (for any reason), autoscaler will most likely scales up another node. It's just a matter of timing who's gonna by first. In general, machine health-checker will be faster as it's purpose is to fix nodes. Whereas, autoscaler's purpose is to dynamically add/remove nodes based on cluster workload.

Reply:
Can you tell me more about the autoscaler and health checker interfering with each other? Is that something that a user should care about? (It sounds like the different processes will eventually self-correct even if they do temporarily collide.)

Reply:
imho this is not something a user should worry about. If they collide in a problematic way, means there's bug we need to fix